esmvaltool/diag_scripts/monitor/multi_datasets.py

```diff
@@ -1862,6 +2209,140 @@ def _get_reference_dataset(self, datasets):
             return ref_datasets[0]
         return None
 
+    def _get_benchmarking_reference(self, datasets):
+        """Extract reference dataset for calculation of benchmarking metric."""
+        variable = datasets[0][self.cfg['group_variables_by']]
+        ref_datasets = [d for d in datasets if
+                        d.get('reference_for_metric', False)]
+
+        if len(ref_datasets) == 1:
+            return ref_datasets[0]
+
+        # try variable attribute "reference_dataset"
+        for dataset in datasets:
+            print(dataset.get('reference_dataset'))
+            print(dataset.get('dataset'))
+            if dataset.get('reference_dataset') == dataset.get('dataset'):
+                ref_datasets = dataset
+                break
+        if len(ref_datasets) != 1:
+            raise ValueError(
+                f"Expected exactly 1 reference dataset for variable "
+                f"'{variable}', got {len(ref_datasets)}")
+        return None
+
+    def _get_benchmark_datasets(self, datasets):
+        """Get dataset to be benchmarked."""
+        variable = datasets[0][self.cfg['group_variables_by']]
+        benchmark_datasets = [d for d in datasets if
+                              d.get('benchmark_dataset', False)]
```

Comment:
Should this really be False? It looks like it's selecting datasets that specify `benchmark_dataset: false`. If correct, please add a comment to avoid confusion.

Reply:
As far as I understand, the Python function `dictionary.get(keyname, value)` returns `value` if the specified key does not exist, so setting `value` to `False` is just to make sure that datasets which do not have a 'benchmark_dataset' key are not selected.